From d3f41642a81f2d208149febe7b3603b7f80f11aa Mon Sep 17 00:00:00 2001
From: Joey Riches <josephriches@gmail.com>
Date: Wed, 15 Aug 2018 23:20:51 +0100
Subject: [PATCH 1/1] Patch out ability to install additional community noises

* This would create a webcontext to prompt install community extension
  noises via dpkg. We already have these packaged and don't support .deb anyway.
  As an additional bonus, this removes the dependency on webkit1.

Signed-off-by: Joey Riches <josephriches@gmail.com>
---
 anoise/preferences.py | 16 +----------
 anoise/preferences.ui | 66 -------------------------------------------
 2 files changed, 1 insertion(+), 81 deletions(-)

diff --git a/anoise/preferences.py b/anoise/preferences.py
index abc9a6b..2b05841 100644
--- a/anoise/preferences.py
+++ b/anoise/preferences.py
@@ -19,8 +19,7 @@
 import gi, os, shutil, webbrowser, subprocess
 from datetime import datetime, timedelta
 gi.require_version('Gtk', '3.0')
-gi.require_version('WebKit', '3.0')
-from gi.repository import Gtk, WebKit
+from gi.repository import Gtk
 # i18n
 import gettext
 gettext.textdomain('anoise')
@@ -43,8 +42,6 @@ class Preferences:
         self.lbl_minutes  = builder.get_object('lbl_minutes')
         self.cb_sleep     = builder.get_object('cb_timesleep')
         self.cb_autostart = builder.get_object('cb_autostart')
-        self.btn_noises   = builder.get_object('btn_show_noises')
-        self.web          = builder.get_object('boxWeb')
         
         # Autostart
         if os.path.isfile(self.AUTOSTART):
@@ -104,17 +101,6 @@ class Preferences:
             self.cb_sleep.set_label(_("Stop in"))
             self._restore_window_size()
     
-    def on_btn_show_noises_clicked(self, widget, data=None):
-        self.btn_noises.hide()
-        web_content = WebKit.WebView()
-        settings = web_content.get_settings()
-        settings.set_property('enable-default-context-menu', False)
-        web_content.set_settings(settings)
-        web_content.open('http://anoise.tuxfamily.org/noises')
-        web_content.connect('navigation-requested', self._on_navigation_requested)
-        self.web.add(web_content)
-        self.web.show_all()
-    
     def _on_navigation_requested(self, view, frame, req):
         uri = req.get_uri()
         if uri and uri.startswith('apt'): # OS installer
diff --git a/anoise/preferences.ui b/anoise/preferences.ui
index 6a706e9..eb59b63 100644
--- a/anoise/preferences.ui
+++ b/anoise/preferences.ui
@@ -166,72 +166,6 @@ sudo shutdown -P 60</property>
             <property name="position">1</property>
           </packing>
         </child>
-        <child>
-          <object class="GtkFrame" id="frame2">
-            <property name="visible">True</property>
-            <property name="can_focus">False</property>
-            <property name="label_xalign">0</property>
-            <property name="shadow_type">none</property>
-            <child>
-              <object class="GtkAlignment" id="alignment2">
-                <property name="visible">True</property>
-                <property name="can_focus">False</property>
-                <property name="left_padding">12</property>
-                <child>
-                  <object class="GtkBox" id="box4">
-                    <property name="visible">True</property>
-                    <property name="can_focus">False</property>
-                    <property name="orientation">vertical</property>
-                    <child>
-                      <object class="GtkButton" id="btn_show_noises">
-                        <property name="label" translatable="yes">Get more online...</property>
-                        <property name="visible">True</property>
-                        <property name="can_focus">True</property>
-                        <property name="receives_default">True</property>
-                        <signal name="clicked" handler="on_btn_show_noises_clicked" swapped="no"/>
-                      </object>
-                      <packing>
-                        <property name="expand">False</property>
-                        <property name="fill">True</property>
-                        <property name="position">0</property>
-                      </packing>
-                    </child>
-                    <child>
-                      <object class="GtkScrolledWindow" id="boxWeb">
-                        <property name="height_request">226</property>
-                        <property name="can_focus">True</property>
-                        <property name="hexpand">True</property>
-                        <property name="vexpand">True</property>
-                        <property name="shadow_type">in</property>
-                        <child>
-                          <placeholder/>
-                        </child>
-                      </object>
-                      <packing>
-                        <property name="expand">False</property>
-                        <property name="fill">True</property>
-                        <property name="position">1</property>
-                      </packing>
-                    </child>
-                  </object>
-                </child>
-              </object>
-            </child>
-            <child type="label">
-              <object class="GtkLabel" id="label2">
-                <property name="visible">True</property>
-                <property name="can_focus">False</property>
-                <property name="margin_bottom">8</property>
-                <property name="label" translatable="yes">Noises:</property>
-              </object>
-            </child>
-          </object>
-          <packing>
-            <property name="expand">False</property>
-            <property name="fill">False</property>
-            <property name="position">2</property>
-          </packing>
-        </child>
         <child>
           <object class="GtkBox" id="box3">
             <property name="visible">True</property>
-- 
2.18.0

